<!-- -*- xml -*- -->
<!--
   Record PAPRAS data automatically in three separate rosbags:
   - cameras images
   - papras information (robot position)
   - diagnostics  output, diagnostics)

   TODO:
   - get regex expressions working for specifying topics
  -->
<launch>
  <arg name="record_all" default="true" />
  <arg name="record_cameras" default="false" />
  <arg name="record_arm1" default="false" />
  <arg name="cam_regex" default="(.*)camera(.*)" />
  <arg name="ROS_WS" default="/home/papras/catkin_ws" />

 <node pkg="rosbag" type="record" name="rosbag_record_all"
       args="record -o $(arg ROS_WS)/bags/papras-all -a"
       if="$(arg record_all)" />

 <node pkg="rosbag" type="record" name="rosbag_record_cam"
       args="record -o $(arg ROS_WS)/bags/papras-cameras -e $(arg cam_regex)"
       if="$(arg record_cameras)" />

 <node pkg="rosbag" type="record" name="rosbag_record_papras"
       args="record -o $(arg ROS_WS)/bags/papras-arm /joint_states /tf
                /arm1_controller/command /arm1_controller/follow_joint_trajectory/goal"
       if="$(arg record_arm1)" />

 <node pkg="rosbag" type="record" name="rosbag_record_diag"
       args="record -o $(arg ROS_WS)/bags/papras-diag /diagnostics /rosout"/>   
</launch>

